conformance/client/src/main/kotlin/com/connectrpc/conformance/client/Client.kt

```diff
@@ -113,63 +159,66 @@ class Client(
-                RequestCompression(10, GzipCompressionPool)
+            if (req.compression == Compression.GZIP) {
+                RequestCompression(0, GzipCompressionPool)
             } else {
                 null
             }
         val compressionPools =
-            if (req.compression == Compression.COMPRESSION_GZIP) {
+            if (req.compression == Compression.GZIP) {
                 listOf(GzipCompressionPool)
             } else {
                 emptyList()
             }
-        return ProtocolClient(
-            httpClient = ConnectOkHttpClient(clientBuilder.build()),
-            ProtocolClientConfig(
-                host = host,
-                serializationStrategy = serializationStrategy,
-                networkProtocol = asNetworkProtocol(req.protocol),
-                getConfiguration = getConfig,
-                requestCompression = requestCompression,
-                compressionPools = compressionPools,
+        val httpClient = clientBuilder.build()
+        return Pair(
+            httpClient,
+            ProtocolClient(
+                httpClient = ConnectOkHttpClient(httpClient),
+                ProtocolClientConfig(
+                    host = host,
+                    serializationStrategy = serializationStrategy,
+                    networkProtocol = req.protocol,
+                    getConfiguration = getConfig,
+                    requestCompression = requestCompression,
+                    compressionPools = compressionPools,
+                ),
             ),
         )
     }
 
-    private fun asNetworkProtocol(protocol: Protocol): NetworkProtocol {
-        return when (protocol) {
-            Protocol.PROTOCOL_CONNECT -> NetworkProtocol.CONNECT
-            Protocol.PROTOCOL_GRPC -> NetworkProtocol.GRPC
-            Protocol.PROTOCOL_GRPC_WEB -> NetworkProtocol.GRPC_WEB
-            else -> throw RuntimeException("unsupported protocol: $protocol")
-        }
-    }
-
-    private fun asOkHttpProtocol(httpVersion: HTTPVersion, useTls: Boolean): okhttp3.Protocol {
+    private fun asOkHttpProtocols(httpVersion: HttpVersion, useTls: Boolean): List<okhttp3.Protocol> {
         return when (httpVersion) {
-            HTTPVersion.HTTP_VERSION_1 -> okhttp3.Protocol.HTTP_1_1
-            HTTPVersion.HTTP_VERSION_2 -> if (useTls) okhttp3.Protocol.HTTP_2 else okhttp3.Protocol.H2_PRIOR_KNOWLEDGE
-            else -> throw RuntimeException("unsupported HTTP version: $httpVersion")
+            HttpVersion.HTTP_1_1 -> listOf(okhttp3.Protocol.HTTP_1_1)
+            HttpVersion.HTTP_2 ->
+                if (useTls) {
+                    // okhttp *requires* that protocols contains HTTP_1_1
+                    // or H2_PRIOR_KNOWLEDGE. So we leave 1.1 in here, but
+                    // expect HTTP/2 to always be used in practice since it
+                    // should be negotiated during TLS handshake,
+                    listOf(okhttp3.Protocol.HTTP_2, okhttp3.Protocol.HTTP_1_1)
```

Comment:
The previous formulation could return a set of protocols that _only_ included `HTTP_2`. However, `okhttp3` doesn't like this and expects `HTTP_1_1` or `H2_PRIOR_KNOWLEDGE` to also be in the list, just in case it encounters a non-TLS request URL.